bazel/scripts/run_integ_tests.sh

```diff
@@ -227,34 +282,43 @@ then
     exit 0
 fi
 
-create_test_targets
+if [[ "${RUN_ALL}" == "true" ]];
+then
+    create_test_targets
+fi
 
-TOTAL_TESTS=${#EXTENDED_TEST_TARGETS[@]}
-# TODO TOTAL_TESTS=$((TOTAL_TESTS + PRECOMMIT))
+TOTAL_TESTS=${#PRECOMMIT_TEST_TARGETS[@]}
+TOTAL_TESTS=$((TOTAL_TESTS + ${#EXTENDED_TEST_TARGETS[@]}))
 
-if [[ "${SKIP_EXTENDED_SETUP_AND_TEARDOWN}" == "false" ]];
+declare -a TEST_BATCH_TO_RUN
+
+if [[ ${#PRECOMMIT_TEST_TARGETS[@]} -gt 0 ]];
 then
-    setup_extended_tests
+    echo "#######################################"
+    echo "PRECOMMIT TESTS"
+    echo "#######################################"
+    TEST_BATCH_TO_RUN=( "${PRECOMMIT_TEST_TARGETS[@]}" )
+    run_test_batch
 fi
 
-for TARGET in "${EXTENDED_TEST_TARGETS[@]}"
-do
-    echo "Starting test ${NUM_RUN}/${TOTAL_TESTS}: ${TARGET}"
-    if run_test "${TARGET}";
+if [[ ${#EXTENDED_TEST_TARGETS[@]} -gt 0 ]];
+then
+    echo "#######################################"
+    echo "EXTENDED TESTS"
+    echo "#######################################"
+    if [[ "${SKIP_EXTENDED_SETUP_AND_TEARDOWN}" == "false" ]];
     then
-        NUM_SUCCESS=$((NUM_SUCCESS + 1))
-        TEST_RESULTS["${TARGET}"]="PASSED"
-    else
-        TEST_RESULTS["${TARGET}"]="FAILED"
+        setup_extended_tests
     fi
-    NUM_RUN=$((NUM_RUN + 1))
-done
 
-if [[ "${SKIP_EXTENDED_SETUP_AND_TEARDOWN}" == "false" ]];
-then
-    teardown_extended_tests
+    TEST_BATCH_TO_RUN=( "${EXTENDED_TEST_TARGETS[@]}" )
+    run_test_batch
+
+    if [[ "${SKIP_EXTENDED_SETUP_AND_TEARDOWN}" == "false" ]];
+    then
+        teardown_extended_tests
+    fi
 fi
-# TODO other tests
 
 print_summary "${NUM_SUCCESS}" "${TOTAL_TESTS}"
```

Comment:
This is not related to the pre-commit tests directly, but to the script output formatting and may be in scope while we are modifying it. When an extended test fails the output can be somewhat misleading due to the tear-down being successful:
![Screenshot from 2022-07-28 09-48-03](https://user-images.githubusercontent.com/34488763/181451256-bcabd159-19dc-45d9-ae45-dc995b8796bb.png)

Perhaps we could make the output of "FAILED" or "FAIL" etc. red and "SUCCESSFUL" etc. green?
Hint: 
- https://stackoverflow.com/questions/5947742/how-to-change-the-output-color-of-echo-in-linux
- https://en.wikipedia.org/wiki/ANSI_escape_code

Reply:
I like this idea, much easier to see the results.